Code/Mantid/Framework/CurveFitting/src/SplineSmoothing.cpp

```diff
@@ -302,29 +309,41 @@ void SplineSmoothing::addSmoothingPoints(const std::set<int> &points,
  * @param row :: The row of spectra to use
  */
 void SplineSmoothing::selectSmoothingPoints(
-    MatrixWorkspace_const_sptr inputWorkspace, size_t row) {
+    MatrixWorkspace_const_sptr inputWorkspace, size_t row, int maxBreaks) {
   std::set<int> smoothPts;
   const auto &xs = inputWorkspace->readX(row);
   const auto &ys = inputWorkspace->readY(row);
 
   int xSize = static_cast<int>(xs.size());
-
+  // if retrienved value is default zero
+  if (maxBreaks == 0) {
+    setProperty("MaxNumberOfBreaks", xs.size());
+    maxBreaks = static_cast<int>(getProperty("MaxNumberOfBreaks"));
+  }
   // number of points to start with
-  int numSmoothPts(M_START_SMOOTH_POINTS);
+  int numSmoothPts(maxBreaks);
 
   // evenly space initial points over data set
   int delta = xSize / numSmoothPts;
-  for (int i = 0; i < xSize; i += delta) {
+  
+  g_log.information() << "delta is: " << delta << std::endl;
```

Comment:
Remember to remove this log message for the final version.

If in this or any algorithm you want to leave some detailed logs that are important to track how the algorithm runs you could put those as "debug" level logs.